doc/instance_wrap.md

```diff
@@ -0,0 +1,405 @@
+# InstanceWrap<T>
+
+This class serves as the base class for [`Napi::ObjectWrap<T>`][] and
+[`Napi::Addon<T>`][].
+
+In the case of [`Napi::Addon<T>`][] it provides the
+methods for exposing functions to JavaScript on instances of an add-on.
+
+As a base class for [`Napi::ObjectWrap<T>`][] it provides the methods for
+exposing instance methods of JavaScript objects instantiated from the JavaScript
+class corresponding to the subclass of [`Napi::ObjectWrap<T>`][].
```

Comment:
Missing the link for:
- `Napi::ObjectWrap<T>`
- `Napi::Addon<T>`
- `Napi::ObjectWrap<T>`